bodo/hiframes/table.py

```diff
@@ -158,6 +158,24 @@ def to_pandas(self, index=None):
         return df
 
 
+def _unify_array_types(typingctx, t1, t2):
```

Comment:
So one issue with just using this function is that the name implies its a "general purpose" array unification function for arrays (which maybe that is the case because if you have conditionals forcing two different integer bit widths this is a reasonable behavior.

However, the use of `_derive_common_key_type` suggests to me that we could have a major issue if this is used outside of a context where key unification is expected (e.g. the data columns of the join). I don't have any objection with this behavior because I think it matches the internal expected behavior, but let's add a TODO to make the `_derive_common_key_type` more generic and follow more "general Bodo rules" for legal type unification.